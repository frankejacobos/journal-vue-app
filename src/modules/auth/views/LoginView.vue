<template>
  <span class="login100-form-title p-b-41"> Ingresar </span>
  <form
    class="login100-form validate-form p-b-33 p-t-5"
    @submit.prevent="onSubmit"
  >
    <div class="wrap-input100 validate-input" data-validate="Enter email">
      <input
        required
        type="text"
        class="input100"
        placeholder="email"
        v-model="userForm.email"
      />
      <span class="focus-input100" data-placeholder="&#xe82a;"></span>
    </div>

    <div class="wrap-input100 validate-input" data-validate="Enter password">
      <input
        required
        type="password"
        class="input100"
        placeholder="password"
        v-model="userForm.password"
      />
      <span class="focus-input100" data-placeholder="&#xe80f;"></span>
    </div>

    <div class="container-login100-form-btn m-t-32">
      <button type="submit" class="login100-form-btn">Login</button>
    </div>

    <div class="container-login100-form-btn m-t-32">
      <router-link :to="{ name: 'register' }">Â¿No tienes cuenta?</router-link>
    </div>
  </form>
</template>

<script>
import { ref } from "vue";
import { useRouter } from "vue-router";
import useAuth from "../composables/useAuth";
import Swal from "sweetalert2";

export default {
  name: "LoginView",
  setup() {
    const router = useRouter();
    const { login } = useAuth();
    const userForm = ref({
      email: "",
      password: "",
    });
    return {
      userForm,
      onSubmit: async () => {
        const { ok, message } = await login(userForm.value);
        if (!ok) {
          Swal.fire({
            icon: "error",
            title: "Oops...",
            text: message,
          });
        } else {
          Swal.fire({
            icon: "success",
            title: "Bienvenido",
            text: message,
          });
          router.push({ name: "no-entry" });
        }
      },
    };
  },
};
</script>

<style></style>
