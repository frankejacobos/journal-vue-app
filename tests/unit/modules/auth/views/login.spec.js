import LoginView from "@/modules/auth/views/LoginView.vue";
import { shallowMount } from "@vue/test-utils";
import createVuexStore from "@/../tests/unit/mocks/mock-store";
import Swal from "sweetalert2";

jest.mock("sweetalert2", () => ({
  fire: jest.fn(),
  showLoading: jest.fn(),
  close: jest.fn(),
}));

describe("Pruebas en el componente LoginView", () => {
  const store = createVuexStore({
    status: "authenticating",
    user: null,
    idToken: null,
    refreshToken: null,
  });
  store.dispatch = jest.fn();
  beforeEach(() => {
    jest.clearAllMocks();
  });
  test("Debe de hacer match con el snapshot", () => {
    const wrapper = shallowMount(LoginView, {
      global: { plugins: [store] },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("Debe de disparar el Swal.fire", async () => {
    store.dispatch.mockReturnValueOnce({ ok: false, message: "Error" });
    const wrapper = shallowMount(LoginView, {
      global: { plugins: [store] },
    });
    await wrapper.find("form").trigger("submit");
    expect(store.dispatch).toHaveBeenCalledWith("auth/login", {
      email: "",
      password: "",
    });
    expect(Swal.fire).toHaveBeenCalledWith({
      icon: "error",
      title: "Oops...",
      text: "Error",
    });
  });
  test("Debe de redireccionar al dashboard (no-entry)", async () => {
    store.dispatch.mockReturnValueOnce({ ok: true, message: "Ok" });
    const wrapper = shallowMount(LoginView, {
      global: { plugins: [store] },
    });
    const [email, password] = wrapper.findAll("input");
    await email.setValue("prueba@gmail.com");
    await password.setValue("123456");
    await wrapper.find("form").trigger("submit");
    expect(store.dispatch).toHaveBeenCalledWith("auth/login", {
      email: "prueba@gmail.com",
      password: "123456",
    });
    expect(Swal.fire).toHaveBeenCalledWith({
      icon: "success",
      title: "Bienvenido",
      text: "Ok",
    });
    expect(wrapper.router.push).toHaveBeenCalledWith({ name: "no-entry" });
  });
});
