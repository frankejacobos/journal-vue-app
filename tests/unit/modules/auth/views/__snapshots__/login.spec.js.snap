// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente LoginView Debe de hacer match con el snapshot 1`] = `
<span class="login100-form-title p-b-41"> Ingresar </span>
<form class="login100-form validate-form p-b-33 p-t-5">
  <div class="wrap-input100 validate-input" data-validate="Enter email"><input required="" type="text" class="input100" placeholder="email"><span class="focus-input100" data-placeholder=""></span></div>
  <div class="wrap-input100 validate-input" data-validate="Enter password"><input required="" type="password" class="input100" placeholder="password"><span class="focus-input100" data-placeholder=""></span></div>
  <div class="container-login100-form-btn m-t-32"><button type="submit" class="login100-form-btn">Login</button></div>
  <div class="container-login100-form-btn m-t-32">
    <router-link-stub to="[object Object]" replace="false" custom="false" ariacurrentvalue="page"></router-link-stub>
  </div>
</form>
`;
